.github/workflows/release-azure-pipelines.yml

```diff
@@ -93,40 +93,268 @@ extends:
+              MaxConcurrency: '50'
+              MaxRetryAttempts: '5'
+              PendingAnalysisWaitTimeoutMinutes: '5'
+          # We need to zip the artifacts for osx before sending to ESRP for signing.
+          - task: ArchiveFiles@2
+            displayName: Codesigning - zip artifacts to send to ESRP
+            condition: startsWith('${{ config.runtime }}', 'osx')
+            inputs:
+              rootFolderOrFile: $(Build.ArtifactStagingDirectory)/azureauth-${{ parameters.version }}-${{ config.runtime }}
+              includeRootFolder: false
+              archiveType: zip
+              archiveFile: $(Build.ArtifactStagingDirectory)/azureauth-${{ parameters.version }}-${{ config.runtime }}.zip
+          - task: EsrpCodeSigning@5
+            displayName: Sign artifacts osx
+            condition: startsWith('${{ config.runtime }}', 'osx')
+            inputs:
+              ConnectedServiceName: $(esrpKVServiceConnection)
+              AppRegistrationClientId: $(SIGNING_AAD_ID)
+              AppRegistrationTenantId: $(SIGNING_TENANT_ID)
+              AuthAKVName: $(AZURE_VAULT)
+              AuthCertName: $(AZURE_VAULT_ESRP_AAD_CERT_NAME)
+              AuthSignCertName: $(AZURE_VAULT_ESRP_REQ_CERT_NAME)
+              FolderPath: $(Build.ArtifactStagingDirectory)
+              Pattern: 'azureauth-${{ parameters.version }}-${{ config.runtime }}.zip'
+              signConfigType: 'inlineSignParams'
+              inlineOperation: |
+                [
+                  {
+                    "KeyCode": "$(SIGNING_KEY_CODE_MAC)",
+                    "OperationCode": "MacAppDeveloperSign",
+                    "ToolName": "sign",
+                    "ToolVersion": "1.0",
+                    "Parameters": {}
+                 },
+                 {
+                  "KeyCode": "$(SIGNING_KEY_CODE_MAC)",
+                  "OperationCode": "SigntoolVerify",
+                  "ToolName": "sign",
+                  "ToolVersion": "1.0",
+                  "Parameters": {}
+                 }
+                ]
+              SessionTimeout: '60'
+              MaxConcurrency: '50'
+              MaxRetryAttempts: '5'
+              PendingAnalysisWaitTimeoutMinutes: '5'
+          - task: ExtractFiles@1
+            displayName: Extract signed artifacts osx
+            condition: startsWith('${{ config.runtime }}', 'osx')
+            inputs:
+              archiveFilePatterns: $(Build.ArtifactStagingDirectory)/azureauth-${{ parameters.version }}-${{ config.runtime }}.zip
+              destinationFolder: $(Build.ArtifactStagingDirectory)/azureauth-${{ parameters.version }}-${{ config.runtime }}
+              cleanDestinationFolder: true
+              overwriteExistingFiles: true
+          # We rename the signed artifacts to avoid conflicts with the unsigned pipeline artifacts from the previous stage.
+          - task: PowerShell@2
+            displayName: Rename signed artifacts

```

Comment:
Is `ExtractFiles@1` going to replace the `azureauth-${{ parameters.version }}-${{ config.runtime }}` file generated during the build task (if any)? Is it possible to have the `destinationFolder` value be `azureauth-${{ parameters.version }}-${{ config.runtime }}-signed` instead and avoid the `PowerShell@2` task?

Reply:
The `extractFiles` task is only required for macos where we need to zip before sending to esrp. However, renaming is required for all the runtimes. So I have kept this as a separate task representing the intentional rename common for all runtimes.

Reply:
That's so weird that mac has this one specific requirement for code-signing! I wonder why that is.

Reply:
It is unfortunate! I tried signing without zipping and got `Incorrect file type used for command -z. It needs to be a .zip or .dmg file.\n`. Later I looked at other pipelines and realized we do need to zip.